webapp/channels/src/components/threading/global_threads/thread_menu/thread_menu.tsx

```diff
@@ -84,6 +88,22 @@ function ThreadMenu({
         unreadTimestamp,
     ]);
 
+    // TODO: This should be in a reusable component but since this menu hasn't been
+    // migrated to the new menu component yet, we'll leave it here for now.
+    const [canPopout, setCanPopout] = useState(false);
+    const checkIfCanPopout = useCallback(async () => {
+        const canPopout = await DesktopApp.canPopout();
```

Comment:
Is this value something we ever expect to change throughout the app's lifecycle, or is it basically set once? I think it might make sense to call this once and store its value somewhere to avoid having popout buttons/menu items flash in shortly after their parent is mounted.

If the web app could subscribe to changes to this value, we could use React 18's [`useSyncExternalStore`](https://react.dev/reference/react/useSyncExternalStore#subscribing-to-a-browser-api) for that, but I don't think there's a way to do that without a new API, so we can't use that yet. It's something to consider in the future though

Reply:
We can just store it in the Desktop App API, we just need to make sure it exists before the app can run. It won't change except when the app is loaded.